pandas/tests/scalar/interval/test_interval.py

```diff
@@ -184,6 +184,184 @@ def test_math_sub(self, closed):
+            interval.right - Timedelta("1 days 00:00:00"),
+            closed=closed,
+        )
+        assert result == expected
+
+    def test_math_add_interval_timestamp_timedelta(self, closed):
+        interval = Interval(
+            Timestamp("1900-01-01"), Timestamp("1900-01-02"), closed=closed
+        )
+        expected = Interval(
+            Timestamp("1900-01-02"), Timestamp("1900-01-03"), closed=closed
+        )
+
+        result = interval + Timedelta("1 days 00:00:00")
+        assert result == expected
+
+        result = interval
+        result += Timedelta("1 days 00:00:00")
+        assert result == expected
+
+        expected = Interval(
+            interval.left + Timedelta("1 days 00:00:00"),
+            interval.right + Timedelta("1 days 00:00:00"),
+            closed=closed,
+        )
+
+        result = interval + Timedelta("1 days 00:00:00")
+        assert result == expected
+
+        result = interval
+        result += Timedelta("1 days 00:00:00")
+        assert result == expected
+
+    def test_math_add_interval_timedelta_timedelta(self, closed):
+        interval = Interval(
+            Timedelta("1 days 00:00:00"), Timedelta("2 days 00:00:00"), closed=closed
+        )
+        expected = Interval(
+            Timedelta("4 days 01:00:00"), Timedelta("5 days 01:00:00"), closed=closed
+        )
+
+        result = interval + Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        result = interval
+        result += Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        expected = Interval(
+            interval.left + Timedelta("3 days 01:00:00"),
+            interval.right + Timedelta("3 days 01:00:00"),
+            closed=closed,
+        )
+
+        result = interval + Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        result = interval
+        result += Timedelta("3 days 01:00:00")
+        assert result == expected
+
+    def test_sub_interval_imedelta_timedelta(self, closed):
+        interval = Interval(
+            Timedelta("1 days 00:00:00"), Timedelta("2 days 00:00:00"), closed=closed
+        )
+        expected = Interval(
+            Timedelta("-3 days +23:00:00"),
+            Timedelta("-2 days +23:00:00"),
+            closed=closed,
+        )
+
+        result = interval - Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        result = interval
+        result -= Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        expected = Interval(
+            interval.left - Timedelta("3 days 01:00:00"),
+            interval.right - Timedelta("3 days 01:00:00"),
+            closed=closed,
+        )
+
+        result = interval - Timedelta("3 days 01:00:00")
+        assert result == expected
+
+        result = interval
+        result -= Timedelta("3 days 01:00:00")
+        assert result == expected
+
+    def test_math_add_interval_timestamp_timestamp(self, closed):
```

Comment:
`test_interval_addition_timestamp`
(left and right bounds have to be the same type)